src/WebJobs.Script/Description/DotNet/DotNetFunctionInvoker.cs

```diff
@@ -189,6 +191,20 @@ public override void OnError(Exception ex)
                 TraceCompilationDiagnostics(exc.Diagnostics, LogTargets.User);
                 throw;
             }
+            catch (CompilerErrorException exc)
+            {
+                TraceOnPrimaryHost("Script compiler error", TraceLevel.Error, _compilationService.GetType().Name, exc);
+
+                // Compiler errors are often sporadic, so we'll attempt to reset the loader here to avoid
+                // caching the compiler error and leaving the function hopelessly broken
+                if (++_compilerErrorCount < 3)
+                {
+                    _functionLoader.Reset();
+
+                    TraceOnPrimaryHost("Function loader reset. Failed compilation result will not be cached.", TraceLevel.Info);
+                }
```

Comment:
Those are tracing messages to give us better visibility when those problems happen. We can flag those as system only.